cpp/src/arrow/python/common.h

```diff
@@ -67,9 +67,12 @@ class ARROW_EXPORT PyAcquireGIL {
 class ARROW_EXPORT OwnedRef {
  public:
   OwnedRef() : obj_(NULLPTR) {}
-
+  OwnedRef(const OwnedRef&) = delete;
+  OwnedRef(OwnedRef&& other) : OwnedRef(other.detach()) {}
   explicit OwnedRef(PyObject* obj) : obj_(obj) {}
 
+  OwnedRef& operator=(const OwnedRef&) = delete;
```

Comment:
Can you use the `ARROW_DISALLOW_COPY_AND_ASSIGN` macro for this, and put it in `private:` scope? Example: https://github.com/apache/arrow/blob/master/cpp/src/arrow/record_batch.h#L161

Reply:
Ah, cool, I didn't know about that.